### DISCUSSION TITLE
Questions about default cache behaviour

### DISCUSSION THREAD

**Discussion by dktapps** [CuyZ/Valinor#501] [General]
In the documentation, there's this line:

> The library provides a cache implementation out of the box, which saves cache entries into the file system.

However, on code inspection, I see that `RuntimeCache` is used by default, which does not store anything on disk as far as I can see. Is the documentation incorrect, or am I missing some part of the code which stores the cache on disk?

(In my application, caching on disk isn't needed, since the application has a long runtime.)

Finally - Is the default cache shared between all instances of `MapperBuilder`, or would a new cache be created each time? Are there any steps I need to make sure that the same default cache is used?

Thanks :)

**Comment by romm** [comment]
Hi @dktapps!

> However, on code inspection, I see that `RuntimeCache` is used by default, which does not store anything on disk as far as I can see. Is the documentation incorrect, or am I missing some part of the code which stores the cache on disk?

Right, `RuntimeCache` is used by default, and should probably have been named differently, perhaps `InMemoryCache` or something similar, as this cache just keeps processed information (functions/interface/class/properties/methods/parameters/types) in memory.

> Finally - Is the default cache shared between all instances of MapperBuilder, or would a new cache be created each time? Are there any steps I need to make sure that the same default cache is used?

If a cache instance with the same directory is given to several `MapperBuilder` instances, they will share the cache items.

Hope that helps!
